File: 404-sum-of-left-leaves/404-sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int leftLeaf(TreeNode* root, bool leaf){
        if(!root) return 0;
        if(!root->left && !root->right && leaf) return root->val;
        int ls = leftLeaf(root->left,true);
        int rs = leftLeaf(root->right,false);
        return ls+rs;
    }
    
    int sumOfLeftLeaves(TreeNode* root) {
        return leftLeaf(root,false);
    }
};

File: 404-sum-of-left-leaves/404-sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def leftLeaf(self,root,leaf):
        if not root: return 0
        if not root.left and not root.right and leaf:
            return root.val
        ls = self.leftLeaf(root.left,True)
        rs = self.leftLeaf(root.right,False)
        return ls+rs
    
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        return self.leftLeaf(root,False)